openreview/venue/venue.py

```diff
@@ -1053,6 +1061,244 @@ def set_SAC_ethics_review_process(self, sac_ethics_flag_duedate=None):
+                            # {
+                            #     "id": "d7cf2650-c1c7-11e8-b568-0800200c9a66",
+                            #     "family_name": "test_instructor_first_name",
+                            #     "given_name": "test_instructor_last_name",
+                            #     "email": "instructor_email@test.com"
+                            # },
+                            # {
+                            #     "id": "7a62f070-c265-11e8-b568-0800200c9a66",
+                            #     "family_name": "test_instructor_2_first_name",
+                            #     "given_name": "test_instrutor_2_last_name",
+                            #     "email": "intructor_2_email@test.com"
+                            # }
+                        ],
+                    },
+                    group_type="ASSIGNMENT",
+                )
+                iThenticate_submission_id = res["id"]
+
+                iThenticate_edge = openreview.api.Edge(
+                    invitation=self.get_iThenticate_plagiarism_check_invitation_id(),
+                    head=submission.id,
+                    tail=iThenticate_submission_id,
+                    label="Created",
+                    weight=-1,
+                )
+
+                iThenticate_edge = self.client.post_edge(iThenticate_edge)
+
+                submission_file_binary_data = self.client.get_attachment(
+                    id=submission.id, field_name="pdf"
+                )
+
+                submission_file_object = io.BytesIO(submission_file_binary_data)
+
+                iThenticate_edge.label = "File Sent"
+                iThenticate_edge = self.client.post_edge(iThenticate_edge)
+
+                try:
+                    res = iThenticate_client.upload_submission(
+                        submission_id=iThenticate_submission_id,
+                        file_data=submission_file_object,
+                        file_name=submission.content["title"]["value"],
+                    )
+                except Exception as err:
+                    iThenticate_edge.label = "Created"
+                    iThenticate_edge = self.client.post_edge(iThenticate_edge)
+
+            else:
+                print(f"Submission {submission.id} already has edge associated with it with label {edges_dict[submission.id][0]['label']}")
+    
+    def handle_ithenticate_upload_error(self):
+
+        if not self.iThenticate_plagiarism_check:
+            raise openreview.OpenReviewException(
+                "iThenticatePlagiarismCheck is not enabled for this venue."
+            )
+
+        iThenticate_client = openreview.api.iThenticateClient(
+            self.iThenticate_plagiarism_check_api_key,
+            self.iThenticate_plagiarism_check_api_base_url,
+        )
+
+        edges = self.client.get_grouped_edges(
+            invitation=self.get_iThenticate_plagiarism_check_invitation_id(),
+            label="Error_PROCESSING_ERROR",
```

Comment:
PROCESSING_ERROR can come from any part of the workflow? either upload a submission or requesting a similarity report?

Reply:
Yes

Reply:
I do think we need to track when this process error happened, during submission upload or similarity report. Based on that we know how to retry.

Could it be two type of error: "Error_upload_PROCESSING_ERROR" and "Error_similarity_PROCESSING_ERROR". 

Then you can upload the submission in the first error or request the similarity report in the second.

Reply:
According to the documentation, both parts can have PROCESSING_ERROR.